src/emailLogin/runMigration.js

```diff
@@ -0,0 +1,88 @@
+
+const ISOMER_USERS = ['isomeradmin', 'rc-davis', 'lamkeewei', 'pallani', 'LoneRifle', 'prestonlimlianjie', 'alexanderleegs', 'lisatjide', 'kwajiehao', 'gweiying', 'seaerchin', 'isomer-demo', 'NatMaeTan', ' jacksonOGP', 'chienlinggg', 'kathleenkhy', 'joshuajunmingt', 'audreytcy', 'yanjunquek', 'chloe-opengovsg', 'shazlithebestie', 'lennardl', 'oliverli', 'taufiq'];
+
+// name of repo
+const REPO = process.argv[2];
+
+const getSiteAndContributors = async (site, dbClient) => {
+  try {
+    const { data: respData } = await axios.get(
+      `https://api.github.com/orgs/${ISOMER_GITHUB_ORG_NAME}/teams/${site}/members`,
+      {
+        headers: {
+          Authorization: `token ${PERSONAL_ACCESS_TOKEN}`,
+        },
+      },
+    );
+    if (!respData) {
+      return;
+    }
+    const contributorNames = respData
+      .map((data) => {
+        const {
+          login,
+        } = data;
+        return login;
+      })
+      .filter((name) => !ISOMER_USERS.includes(name));
+    // get user information
+    const contributorQuery = `SELECT * FROM "users" WHERE ${contributorNames.map((githubId) => `github_id='${githubId}'`).join(' OR ')};`;
+    const userData = (await dbClient.query(contributorQuery)).rows;
+    console.log(userData);
+
+    // get repo information
+    const repoQuery = `SELECT sites.id, sites.job_status, sites.site_status, repos.name FROM "sites" JOIN "repos" ON sites.id = repos.site_id WHERE repos.name='${site}';`;
+    const repoData = (await dbClient.query(repoQuery)).rows;
+    console.log(repoData);
+
+    // get list of whitelisted domains
+    const whitelistQuery = 'SELECT * FROM "whitelist" WHERE expiry IS NULL;';
+    const whitelistedDomains = (await dbClient.query(whitelistQuery)).rows.map((whitelistData) => whitelistData.email);
+    console.log(whitelistedDomains);
+
+    const repoId = repoData[0].id;
+    const siteMemberValues = [];
+    userData.forEach((user) => {
+      const userId = user.id;
+      const userType = whitelistedDomains.filter((domain) => user.email.endsWith(domain)).length > 0 ? 'ADMIN' : 'CONTRIBUTOR';
+      siteMemberValues.push(`(${userId}, ${repoId}, '${userType}')`);
+    });
+    const insertQuery = `INSERT INTO "site_members" (user_id, site_id, role) VALUES\n${siteMemberValues.join(',\n')};`;
+    await dbClient.query(insertQuery);
+    console.log(insertQuery);
+    try {
+      if (!fs.existsSync('./emailMigrationData')) {
+        fs.mkdirSync('./emailMigrationData');
+      }
+      const dirPath = `./emailMigrationData/${site}`;
+      if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath);
+      }
+      fs.writeFileSync(`${dirPath}/contributors.txt`, userData.map((userInfo) => JSON.stringify(userInfo)).join('\n'));
+      fs.writeFileSync(`${dirPath}/repos.txt`, JSON.stringify(repoData[0]));
+      fs.writeFileSync(`${dirPath}/insertQueries.txt`, insertQuery);
+      // file written successfully
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
wouldn't this be caught by the overarching catch below?

Reply:
I put this in a separate catch block because this only wraps the log creation - it would still be successful without it, what do you think?

Reply:
this is a case where it works but it runs counter to clarity; i think what would be good here is: 

1. determine if email part is important enough to fail (in here, we don't so clearly not AHHA)
2. shift some stuff out (thinking the second `try` block) so that it's clearer what the block does

Reply:
Sure, that makes sense, moved it out in https://github.com/isomerpages/isomer-conversion-scripts/pull/56/commits/ad8f51847323111bd76b5136823e56138970caf8